From 1f69fb29312271612312fbaca88b089256b6835e Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Mon, 9 Feb 2015 10:40:40 +0800
Subject: [PATCH] HDMI: fix filter feature SUPPORT_TMDS_600M error: 	If soc
 is not support feature SUPPORT_TMDS_600M, 	or current EDID information is
 not support the 	feature, we should not show the corresponding 
 option, such as 3840x2160p-60.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c b/drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c
index 1bd26e7e7f4f..56dd6aa0705f 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c
@@ -447,10 +447,10 @@ static void hdmi_sort_modelist(struct hdmi_edid *edid, int feature)
 			vic = modelist->vic & HDMI_VIC_MASK;
 			if (vic == hdmi_mode[i].vic ||
 			    vic == hdmi_mode[i].vic_2nd) {
-				if ((feature & SUPPORT_TMDS_600M) == 0 &&
-				    !(modelist->vic & HDMI_VIDEO_YUV420) &&
-				    hdmi_mode[i].mode.pixclock > 340000000 &&
-				    edid->maxtmdsclock < 340000000)
+				if ((((feature & SUPPORT_TMDS_600M) == 0 &&
+				    !(modelist->vic & HDMI_VIDEO_YUV420)) ||
+				    edid->maxtmdsclock < 340000000) &&
+				    hdmi_mode[i].mode.pixclock > 340000000)
 					continue;
 				if ((feature & SUPPORT_4K) == 0 &&
 				    hdmi_mode[i].mode.xres >= 3840)
-- 
2.35.3

